prototypes/matrix_free_navier_stokes/matrix_free_navier_stokes.cc

```diff
@@ -0,0 +1,2420 @@
+                    std::sin(a * y) * std::cos(a * x) +
+                  2 * a * std::sin(a * x) * std::sin(a * x) * std::sin(a * y) *
+                    std::sin(a * y) * std::sin(a * y) * std::cos(a * x) *
+                    std::cos(a * x) * std::cos(a * y) +
+                  (a * std::sin(a * x) * std::sin(a * x) * std::sin(a * y) *
+                     std::sin(a * y) -
+                   a * std::sin(a * y) * std::sin(a * y) * std::cos(a * x) *
+                     std::cos(a * x)) *
+                    std::sin(a * x) * std::sin(a * x) * std::sin(a * y) *
+                    std::cos(a * y));
+        }
+      else
+        {
+          return 0;
+        }
+    }
+
+  if constexpr (dim == 3)
+    {
+      const double z = p[2];
+      if (component == 0)
+        {
+          return (2 * a * a * (2 - 3 * std::cos(2 * a * x)) * std::sin(a * y) *
+                    std::sin(a * z) * std::cos(a * y) * std::cos(a * z) +
+                  a *
+                    (2 * std::cos(a * y) * std::cos(a * y) -
+                     std::cos(a * z) * std::cos(a * z)) *
+                    std::sin(a * x) * std::sin(a * x) * std::sin(a * x) *
+                    std::sin(a * y) * std::sin(a * y) * std::sin(a * z) *
+                    std::sin(a * z) * std::cos(a * x) +
+                  2 * x);
+        }
+      else if (component == 1)
+        {
+          return (2 * a * a * (2 - 3 * std::cos(2 * a * y)) * std::sin(a * x) *
+                    std::sin(a * z) * std::cos(a * x) * std::cos(a * z) +
+                  a *
+                    (2 * std::cos(a * x) * std::cos(a * x) -
+                     std::cos(a * z) * std::cos(a * z)) *
+                    std::sin(a * x) * std::sin(a * x) * std::sin(a * y) *
+                    std::sin(a * y) * std::sin(a * y) * std::sin(a * z) *
+                    std::sin(a * z) * std::cos(a * y) +
+                  2 * y);
+        }
+      else
+        {
+          return (4 * a * a * (3 * std::cos(2 * a * z) - 2) * std::sin(a * x) *
+                    std::sin(a * y) * std::cos(a * x) * std::cos(a * y) +
+                  2 * a *
+                    (std::cos(a * x) * std::cos(a * x) +
+                     std::cos(a * y) * std::cos(a * y)) *
+                    std::sin(a * x) * std::sin(a * x) * std::sin(a * y) *
+                    std::sin(a * y) * std::sin(a * z) * std::sin(a * z) *
+                    std::sin(a * z) * std::cos(a * z) +
+                  2 * z);
+        }
+    }
+}
+
+template <int dim>
+double
+FullSourceTerm<dim>::value(const Point<dim> & p,
+                           const unsigned int component) const
+{
+  return value<double>(p, component);
+}
+
+// Matrix-free helper function
+template <int dim, typename Number>
+VectorizedArray<Number>
+evaluate_function(const Function<dim> &                      function,
+                  const Point<dim, VectorizedArray<Number>> &p_vectorized)
+{
+  VectorizedArray<Number> result;
```

Comment:
I am not sure about the standard that we have for comments in prototypes, but I would add comments here to explain why it is unnecessary to initialize the VectorizedArray size.